cmd/snap-confine/snap-confine.c

```diff
@@ -88,6 +89,26 @@ int main(int argc, char **argv)
 		die("need to run as root or suid");
 	}
 #endif
+
+	char *snap_context __attribute__ ((cleanup(sc_cleanup_string))) = NULL;
+	const char *snap_name = getenv("SNAP_NAME");
```

Comment:
Thanks!

Can you find the other place that cares about SNAP_NAME and either also verify it there or obtain the value once and pass it around please.